cmd/geth/main.go

```diff
@@ -490,7 +489,23 @@ func startNode(ctx *cli.Context, stack *node.Node) {
 		}
 	}
 	if !ctx.GlobalBool(utils.VersionCheckFlag.Name) {
-		blockchain_parameters.SpawnCheck()
+		var callerFactory func() (vm.EVMRunner, error)
+		if ctx.GlobalString(utils.SyncModeFlag.Name) == "full" || ctx.GlobalString(utils.SyncModeFlag.Name) == "fast" {
```

Comment:
I don't like this, but I don't see a way to avoid it entirely.  At first I thought to combine this with the block above that also checks full/light, but that involves its own nested checks.

However, I have a feeling that this `"full" || "fast"` may bite us in the future, in the same way that https://github.com/celo-org/celo-blockchain/blob/master/cmd/utils/flags.go#L1224 is now wrong since we added the `lightest` sync mode.  Perhaps what I would suggest is having a local var higher up like:
```
fullNode := ctx.GlobalString(utils.SyncModeFlag.Name) == "full" || ctx.GlobalString(utils.SyncModeFlag.Name) == "fast"
```

and then using that below for the two if's.  This will likely cause a conflict later on when `snap` sync mode is brought in from upstream, but at least such a conflict would prevent us from overlooking that the new condition added in this PR goes out of date.  Then again, I guess if something like that did happen, we would probably get a `Fatalf` here which would get detected by one of the end-to-end sync tests.

So I don't know, whatever you think best.

Reply:
done!